src/plugins/discover/public/application/main/hooks/use_saved_search.ts

```diff
@@ -151,36 +151,56 @@ export const useSavedSearch = ({
   }>({});
 
   /**
-   * This part takes care of triggering the data fetching by creating and subscribing
-   * to an observable of various possible changes in state
+   * handler emitted by `timefilter.getAutoRefreshFetch$()`
+   * to notify when data completed loading and to start a new autorefresh loop
    */
-  useEffect(() => {
-    /**
-     * handler emitted by `timefilter.getAutoRefreshFetch$()`
-     * to notify when data completed loading and to start a new autorefresh loop
-     */
-    const setAutoRefreshDone = (fn: AutoRefreshDoneFn | undefined) => {
-      refs.current.autoRefreshDone = fn;
-    };
-    const fetch$ = getFetch$({
-      setAutoRefreshDone,
+  const setAutoRefreshDone = useCallback((fn: AutoRefreshDoneFn | undefined) => {
+    refs.current.autoRefreshDone = fn;
+  }, []);
+
+  /**
+   * Observable that allows listening for when fetches are triggered
+   */
+  const fetch$ = useMemo(
+    () =>
+      getFetch$({
+        setAutoRefreshDone,
+        data,
+        main$,
+        refetch$,
+        searchSessionManager,
+        searchSource,
+        initialFetchStatus,
+      }).pipe(
+        filter(() => validateTimeRange(timefilter.getTime(), services.toastNotifications)),
+        tap(() => inspectorAdapters.requests.reset()),
+        share()
+      ),
+    [
       data,
+      initialFetchStatus,
+      inspectorAdapters.requests,
       main$,
       refetch$,
       searchSessionManager,
       searchSource,
-      initialFetchStatus,
-    });
+      services.toastNotifications,
+      setAutoRefreshDone,
+      timefilter,
+    ]
+  );
```

Comment:
those deps will soon be gone, just saying ... with the whole file saved from a hook ... https://github.com/elastic/kibana/pull/148614